KenticoCloud.Delivery/StrongTyping/DeliveryItemsFeed.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace KenticoCloud.Delivery
+{
+    /// <summary>
+    /// Represents a feed that can be used to retrieve strongly typed content items from Kentico Cloud Delivery API in smaller batches.
+    /// </summary>
+    /// <typeparam name="T">The type of content items in the feed.</typeparam>
+    public class DeliveryItemsFeed<T> : IEnumerable<T>
+    {
+        internal delegate Task<DeliveryItemsFeedResponse<T>> GetFeedResponse(string continuationToken);
+
+        private DeliveryItemsFeedResponse<T> _lastResponse;
+        private readonly GetFeedResponse _getFeedResponseAsync;
+
+        /// <summary>
+        /// Indicates whether there are more batches to fetch.
+        /// </summary>
+        public bool HasMoreResults => _lastResponse == null || !string.IsNullOrEmpty(_lastResponse.ContinuationToken);
+
+        /// <summary>
+        /// Gets the URL used to retrieve responses in this feed for debugging purposes.
+        /// </summary>
+        public string ApiUrl { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeliveryItemsFeed{T}"/> class.
+        /// </summary>
+        /// <param name="getFeedResponseAsync">Function to retrieve next batch of content items.</param>
+        /// <param name="requestUrl">URL used to retrieve responses for this feed.</param>
+        internal DeliveryItemsFeed(GetFeedResponse getFeedResponseAsync, string requestUrl)
+        {
+            _getFeedResponseAsync = getFeedResponseAsync;
+            ApiUrl = requestUrl;
+        }
+
+        /// <summary>
+        /// Retrieves the next feed batch if available.
+        /// </summary>
+        /// <returns>Instance of <see cref="DeliveryItemsFeedResponse{T}"/> class that contains a list of strongly typed content items.</returns>
+        public async Task<DeliveryItemsFeedResponse<T>> FetchNextBatchAsync()
+        {
+            if (HasMoreResults)
+            {
+                _lastResponse = await _getFeedResponseAsync(_lastResponse?.ContinuationToken);
+            }
+
+            return _lastResponse;
+        }
+
+        /// <summary>
+        /// Returns an enumerator that iterates through the collection of content items from all batches.
+        /// </summary>
+        public IEnumerator<T> GetEnumerator()
```

Comment:
@sutrkiller, I'm not familiar with the new endpoint, but isn't this a bit dangerous? I mean - how many API calls can you 'accidentally' make by iterating through all batches? Say you have 100k content items, how would this work?

Reply:
Yes, it is a little dangerous as it can end up in more API calls. This endpoint does not return modular content and will be used only by a fraction of users. This is necessary for current users who will need to transfer to this endpoint with a least amount of changes to their codebase.
Compare: 
`foreach (var item in await _client.GetItemsAsync())`
vs 
`foreach (var item in _client.GetItemsFeed()`

Reply:
I'm honestly not sure this is a good idea. Saying that only a few people will use it (for now) is not very satisfactory. People will eventually use it in some way or the other. I'm not that worried about the size of the response, but by the number of queries it can make. 

Image developer not familiar of what this does - they call this method on their dev environment (where they have only a handful of items) and everything works nicely, so they move their code to prod and everything fails because now it doesn't make 1-2 HTTP requests, it makes 1 000. Depending on the implementation you can pretty much kill your server for everyone. 

If we want to use this, we should be **very explicit** of what it does. I mean, even if you had to make 'just' 20 requests to get all items, it could still be a relatively long wait - 20 * average response time of 50ms = 1 sec - just because devs don't know what this does under the hood.

Reply:
We decided to remove the potentially dangerous code.